### DISCUSSION TITLE
DownloadHelper CoApp (vdhcoapp-win7-x86_64-installer.exe) has a virus per ZoneAlarm Free Antivirus + Firewall version: 15.8.043.18324

### DISCUSSION THREAD

**Discussion by FrickinFruckinFranken** [aclap-dev/video-downloadhelper#198] [Help and Q&A]
### Your question

Are there any other reports of a virus detection in 'vdhcoapp-win7-x86_64-installer.exe'?
I can also upload a ZoneAlarm log / log excerpt, however, when I reviewed the ZA log myself, I found the plethora of entries a tad overwhelming ... I was unable to isolate out a relevant log chunk to copy and am hesitant to post the entire ZA log - suggestions?
FYI - upon discovering the supposed virus during CoApp 2.0.10 (? - version not self-evident in filename, as has been past practice)  install, the ' ... installer.exe' was 'treated' (i.e, quarantined then ultimately deleted) by ZA, thus the previous CoApp (1.6.3) residuals are the only components still evident (see PNG image pasted in below).
Thanks, FFF

### Addon & CoApp version and details

Firefox Browser Extended Support Release 115.8.0esr (64-bit)
Windows 7 Professional SP1
64-bit
https://github.com/aclap-dev/vdhcoapp/releases/latest/download/vdhcoapp-win7-x86_64-installer.exe
<code>
Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.

C:\Users\Frankenstien>cd C:\

C:\

C:\>cd C:\Program Files\net.downloadhelper.coapp\bin\

C:\Program Files\net.downloadhelper.coapp\bin>vdhcoapp.exe --info
'vdhcoapp.exe' is not recognized as an internal or external command,
operable program or batch file.

C:\Program Files\net.downloadhelper.coapp\bin>
<code>
![image](https://github.com/aclap-dev/video-downloadhelper/assets/160812282/18204106-41c9-46e4-83ca-7a3c84865800)

**Comment by paulrouget** [comment]
Every time we release a new coapp, it's the same story. Some antivirus go crazy and flag our binaries.

Our official answer is this:

In some rare cases, antiviruses flag the CoApp. This is a false positive. You don't have to trust our word for it: this software is OpenSource, anyone can look at the source code. See for yourself: https://github.com/aclap-dev/vdhcoapp/. You can build it yourself if you have no trust in our builds.

_If you encounter an antivirus issue, you need to allow the .exe files to be executed, and also allow TCP connections from these .exe files._

And allow us to be blunt here: the only antivirus that is not a scam is [Microsoft Defender](https://learn.microsoft.com/en-us/microsoft-365/security/defender-endpoint/microsoft-defender-antivirus-windows?view=o365-worldwide). It is is more advanced, it's built in Windows, **by Microsoft**, and it's free. I worked at Mozilla on Firefox for 10+ years, and Avast, Norton and all have just been a pain to deal with, with literally zero benefits for the user compared to the builtin Windows antivirus. [Here is another Mozilla developer explaining why all antivirus are bad (except the one built in Windows)](https://robert.ocallahan.org/2017/01/disable-your-antivirus-software-except.html).

Our recommendation: delete your antivirus and make sure to enable [windows defender](https://learn.microsoft.com/en-us/microsoft-365/security/defender-endpoint/microsoft-defender-antivirus-windows?view=o365-worldwide) (and it's probably already active anyway).

**Reply by FrickinFruckinFranken** [reply]
Thank you **Paul** for your prompt & detailed Reply ... appreciated.
I have Microsoft Security Essentials also loading in my systray ... would that be the Win 7 equivalent to Windows Defender on newer MS OS'?
**FFF**

**Reply by GrampaWildWilly** [reply]
I believe it is.

What follows isn't particularly directed at @FrickinFruckinFranken, just a general comment.  I've seen behavior in at least one other forum that parallels some of what I've see in our forum here on GitHub as well as previously in our old forum on Google Groups.

Piling on after Paul's comment, I find it quite ridiculous how a bunch of people get up in arms when some antivirus program or some antivirus rating site flags something.  Antivirus is just a computer application.  Like any application, it is prone to bugs.  People seem to have an almost religious belief in their antivirus that they would never accord any other software.  Plus consider the totality of computer software in the world.  How many applications issue new releases every day?  Every hour?  Do you honestly believe an antivirus company keeps up with those?  They rely on their users telling them about positives so they can investigate whether they are false.  Then some future version of their virus signatures tells their antivirus engine not to flag application such-and-such.  Which is good until that application comes out with their next release.  Then you just rinse & repeat.

Meanwhile, you have folks here who have used VDH for a while, sometimes years, maybe even more than 10 years.  They have personal experience with the product that has established a track record that it is virus-free.  But suddenly some antivirus flags a new release & people chuck the evidence of their own long experience out the window.  If you got the software from the authorized site, then your experience forces you to conclude it has no viruses.  If your antivirus/firewall suddenly flags it, use some common sense.  Make an exception for the new release in both your antivirus & your firewall.  The antivirus I used to use on Windows 7 would flag a small circle of .bat files I updated every day.  The same files, every time I updated them.  Apparently, my changes -- my changes now, not some unknown external influence, my own changes -- caused the antivirus to flag the files over & over & over.  Now that I'm on W11, the same behavior on my part hasn't caused Windows Defender to raise an eyebrow.  It also hasn't flagged any of the several releases of the CoApp we've had in the 2 months & counting that I've been running W11.  I think it has showed me a message 3 or 4 times that something was unknown to it.  I just told it to run the thing anyway, because I knew where I had gotten it from & I trusted the source.

I suppose if you're going to be a fine upstanding netizen, you'll contact your antivirus vendor to inform them of their false positive.  I'm too lazy.  I make the exception in the antivirus & get on with using the application I've just told the antivirus about.  I'm sure Microsoft is none the poorer for my egregious laziness.

**Reply by FrickinFruckinFranken** [reply]
Thank you **GWW** for the first sentence in your 1533PST22FEB24 Reply.
**FFF**
